src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -173,3 +173,249 @@ TEST(map_test, incomplete_map_test)
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("5", result["timestamp.seconds"]);
+    ASSERT_EQ("MST", result["timestamp.timezone"]);
+}
+
+TEST(hlpTimestampTests, ruby)
+{
+    static const char *logQl  = "[<timestamp>]";
+    static const char *rubyTs = "[Mon Jan 02 15:04:05 -0700 2006]";
+
+    auto parseOp = getParserOp(logQl);
+    auto result  = parseOp(rubyTs);
+    ASSERT_EQ("2006", result["timestamp.year"]);
+    ASSERT_EQ("1", result["timestamp.month"]);
+    ASSERT_EQ("2", result["timestamp.day"]);
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("5", result["timestamp.seconds"]);
+    ASSERT_EQ("-0700", result["timestamp.timezone"]);
+}
+
+TEST(hlpTimestampTests, stamp)
+{
+    static const char *logQl        = "[<timestamp>]";
+    static const char *stampTs      = "[Jan 2 15:04:05]";
+    static const char *stampmilliTs = "[Jan 2 15:04:05.000]";
+    static const char *stampmicroTs = "[Jan 2 15:04:05.000000]";
+    static const char *stampnanoTs  = "[Jan 2 15:04:05.000000000]";
+
+    auto parseOp = getParserOp(logQl);
+    auto result  = parseOp(stampTs);
+    ASSERT_EQ("1", result["timestamp.month"]);
+    ASSERT_EQ("2", result["timestamp.day"]);
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("5", result["timestamp.seconds"]);
+
+    auto resultMillis = parseOp(stampmilliTs);
+    ASSERT_EQ("1", resultMillis["timestamp.month"]);
+    ASSERT_EQ("2", resultMillis["timestamp.day"]);
+    ASSERT_EQ("15", resultMillis["timestamp.hour"]);
+    ASSERT_EQ("4", resultMillis["timestamp.minutes"]);
+    ASSERT_EQ("5", resultMillis["timestamp.seconds"]);
+
+    auto resultMicros = parseOp(stampmicroTs);
+    ASSERT_EQ("1", resultMicros["timestamp.month"]);
+    ASSERT_EQ("2", resultMicros["timestamp.day"]);
+    ASSERT_EQ("15", resultMicros["timestamp.hour"]);
+    ASSERT_EQ("4", resultMicros["timestamp.minutes"]);
+    ASSERT_EQ("5", resultMicros["timestamp.seconds"]);
+
+    auto resultNanos = parseOp(stampnanoTs);
+    ASSERT_EQ("1", resultNanos["timestamp.month"]);
+    ASSERT_EQ("2", resultNanos["timestamp.day"]);
+    ASSERT_EQ("15", resultNanos["timestamp.hour"]);
+    ASSERT_EQ("4", resultNanos["timestamp.minutes"]);
+    ASSERT_EQ("5", resultNanos["timestamp.seconds"]);
+}
+
+TEST(hlpTimestampTests, Unix)
+{
+    static const char *logQl  = "[<timestamp>]";
+    static const char *unixTs = "[Mon Jan 2 15:04:05 MST 2006]";
+
+    auto parseOp = getParserOp(logQl);
+    auto result  = parseOp(unixTs);
+    ASSERT_EQ("2006", result["timestamp.year"]);
+    ASSERT_EQ("1", result["timestamp.month"]);
+    ASSERT_EQ("2", result["timestamp.day"]);
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("5", result["timestamp.seconds"]);
+}
+
+TEST(hlpTimestampTests, specific_format)
```

Comment:
If we implement the above tests with an specific format, this can be reused to test the four formats without specifications

Reply:
done